src/Views/BaseElement.elm

```diff
@@ -0,0 +1,143 @@
+    , selectedImpact : Definition
+    , selectElement : element -> Autocomplete element -> msg
+    , toString : element -> String
+    , update : BaseElement element quantity -> BaseElement element quantity -> msg
+    }
+
+
+view : Config element quantity msg -> List (Html msg)
+view { baseElement, db, defaultCountry, delete, disableCountry, disableQuantity, excluded, impact, quantityView, selectedImpact, selectElement, toString, update } =
+    let
+        updateEvent =
+            update baseElement
+
+        deleteEvent =
+            delete baseElement.element
+
+        autocompleteState =
+            db.elements
+                |> List.filter (\component -> not (List.member component excluded))
+                |> AutocompleteSelector.init toString
+    in
+    [ span [ class "QuantityInputWrapper" ]
+        [ quantityView
+            { disabled = disableQuantity
+            , quantity = baseElement.quantity
+            , onChange =
+                \maybeQuantity ->
+                    case maybeQuantity of
+                        Just quantity ->
+                            updateEvent { baseElement | quantity = quantity }
+
+                        _ ->
+                            updateEvent baseElement
+            }
+        ]
+    , selectorView baseElement.element toString (selectElement baseElement.element autocompleteState)
+    , db.countries
+        |> List.sortBy .name
+        |> List.map
+            (\{ code, name } ->
+                option
+                    [ selected (Maybe.map .code baseElement.country == Just code)
+                    , value <| Country.codeToString code
+                    , disabled disableCountry
+                    ]
+                    [ text name ]
+            )
+        |> (::)
+            (option
+                [ value ""
+                , selected (baseElement.country == Nothing)
+                ]
+                [ text <| "Par défaut (" ++ defaultCountry ++ ")" ]
+            )
+        |> select
+            [ class "form-select form-select CountrySelector"
+            , onInput
+                (\val ->
+                    updateEvent
+                        { baseElement
+                            | country =
+                                if val /= "" then
+                                    Country.codeFromString val
+                                        |> (\countryCode -> Country.findByCode countryCode db.countries)
+                                        |> Result.toMaybe
+
+                                else
+                                    Nothing
+                        }
+                )
+            ]
+    , span [ class "text-end ImpactDisplay fs-7" ]
+        [ impact
+            |> Format.formatImpact selectedImpact
+        ]
+    , deleteItemButton (List.length excluded == 1) deleteEvent
+    ]
+
+
+deleteItemButton : Bool -> msg -> Html msg
+deleteItemButton disable event =
```

Comment:
Nit: how about typing this `Bool` explicitly as `{ disabled : Bool }`, so it can be called as `deleteItemButton { disabled = List.length excluded == 1 } deleteEvent`?